From e39deb1575daa8d723afeae56a7b4391a432d900 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 26 Aug 2018 22:43:49 +0300
Subject: [PATCH] Add BTL package support finalize

Now sectors-be.bin packed by post-iamge.sh comes from
new btl package hence no need to use local BTL source copy

I also updated *-distclean goal so transition between
devices will be smooth.
Finally added btl make dependency with uboot.

After this new BTL is ready.

TODO: remove local BTL source copy from this repository
---

diff --git a/Makefile b/Makefile
index fa9beba..e0f54f5 100644
--- a/Makefile
+++ b/Makefile
@@ -190,7 +190,8 @@
 		uboot-custom linux-custom armcp-custom \
 		spdlog-custom specs-custom specs_external-custom \
 		host-uboot-test-hooks-custom host-tbot-custom \
-		thermal-custom preboot-custom host-fw_loader-custom; \
+		thermal-custom preboot-custom host-fw_loader-custom \
+		btl-custom; \
 	fi
 
 .PHONY: hl-distclean
diff --git a/board/habanalabs/common/post-image.sh b/board/habanalabs/common/post-image.sh
index 1798912..1b1f588 100755
--- a/board/habanalabs/common/post-image.sh
+++ b/board/habanalabs/common/post-image.sh
@@ -47,15 +47,6 @@
 	# Used by TRACE32 (Lauterbach JTAG)
 	dd status=none conv=sync bs=4 if=u-boot.bin						 > ${HL_DEV}-u-boot.bin
 
-	### Construct bin file
-	# BTL likes these to be 16B aligned in size
-	dd status=none conv=sync bs=16 if=preboot.bin						 > ${HL_DEV}-preboot.bin
-	# prepare BTL image
-	BTL_CMD_ARGS="${BTL_CMD_ARGS} SPL_SIZE=$(printf "0x%X\n" `stat -c "%s" ${SPL_BINFILE}`)"
-	cp -a ${BOARD_BASE}/../common/btl .
-	make -C btl clean all ${BTL_CMD_ARGS}
-	cp btl/sectors-be.bin .
-
 	# Metadata
 	dd status=none conv=sync bs=64K count=1 if=/dev/zero				 > ${HL_DEV}-pkey-hash.bin
 	dd status=none conv=sync bs=64K count=1 if=/dev/zero				 > ${HL_DEV}-ver.bin
diff --git a/package/btl/btl.mk b/package/btl/btl.mk
index 0cda194..248cb63 100644
--- a/package/btl/btl.mk
+++ b/package/btl/btl.mk
@@ -2,10 +2,15 @@
 BTL_SITE = btl
 BTL_LICENSE = unclear
 
-BTL_DEPENDENCIES = preboot
+BTL_DEPENDENCIES = preboot uboot
 
 define BTL_BUILD_CMDS
 	$(TARGET_MAKE_ENV) $(MAKE) -C $(@D) all
 endef
 
+BTL_INSTALL_IMAGES = YES
+define BTL_INSTALL_IMAGES_CMDS
+	cp -dpf $(@D)/sectors-be.bin $(BINARIES_DIR)/
+endef
+
 $(eval $(generic-package))
